steam_wrapper/steam_wrapper.py

```diff
@@ -246,10 +248,38 @@ def configure_shared_library_guard():
         if mode > 1:
             os.environ["LD_BIND_NOW"] = "1"
 
+def setup_proton_extensions(current_info):
+    proton_dest = Path('.local/share/Steam/compatibilitytools.d')
+    proton_dest.mkdir(parents=True, exist_ok=True)
+
+    # Copy extensions if they exist
+    for proton_ext in Path('/app/proton').iterdir():
+        if not proton_ext.is_dir():
+            continue
+
+        proton_ext_id = f'com.valvesoftware.Steam.Proton.{proton_ext.name}'
+        proton_ext_dest = proton_dest / proton_ext.name
+        proton_ext_commit_hash = current_info["app-extensions"][proton_ext_id]
+        proton_ext_commit_file = proton_ext_dest / '.extension-commit'
+
+        if proton_ext_dest.exists():
+            if proton_ext_commit_file.exists():
+                with proton_ext_commit_file.open() as fp:
+                    if fp.read() == proton_ext_commit_hash:
+                        continue
+
+        print(f"Copying extension {proton_ext}")
+        if proton_ext_dest.exists():
```

Comment:
If you want to avoid copying every time, just write a manifest during build time (sha256 sums of every file) and check if manifest has changed during runtime. It's unnecessary to actually verify hashes if you want the directories to be largely mutable.

Reply:
This has been solved already above by using flatpak commit hashes right above thanks to @refi64's help though?.. It doesn't actually check files themselves, it just checks if the file hash is the same as when it was first copied. This allows for both mutability, speed and still keeping the copy in tandem with the extension.

Reply:
I think the hash resets every time there's any change within this app but could be wrong.

Reply:
I guess this is fine. If we hit issues, let's improve later.

Reply:
Some explanation behind this:

It basically came down to that checking individual sha sums of these files often ended up slowing things down more than they helped, and I doubt there's going to be rebuilds unless something actually changed. Manual sha comparisons also introduce the potential for random stale files if something *does* change, and the code would be a bit more complex than expected since Proton distributions often have default wineprefixes with e.g. recursive symlinks.

TL;DR: not really worth it for now...

Reply:
Afaik Flathub always recompiles entire app whenever *anything* changes. Do you think it makes sense eg systemd version bump results in wiping and recreating the Proton data? I'm not entirely sure if this happens with app extensions but would expect so.

Reply:
Also as said, build-time hashing does not really slow things down for user. In runtime you can just copy over manifest and go no-op if manifest under /app and under deploy dir are identical (without actually verifying files).

Reply:
Whenever...anything changes? The commit of the extension is checked, and as the original app is used as a runtime, the only times the commit should change would be when the extension itself is rebuilt.

Reply:
You're confused app extension is built as part of the application when done the way described in this PR. Any app change results in the extension to be rebuilt too afaik.

Reply:
@nanonyme I assumed the the extension's manifests are here just for example, and later they will be moved to separate repositories. Are they going to be maintained here, as a part of the application?

Reply:
I assumed this was more or less the final plan. I don't know any context beyond thos PR and related ticket.